src/dataframe/dataframe.jl

```diff
@@ -147,7 +149,8 @@ function DataFrame(; kwargs...)
 end
 
 function DataFrame(columns::AbstractVector,
-                   cnames::AbstractVector{Symbol}=gennames(length(columns));
+                   cnames::AbstractVector{Symbol}=gennames(length(columns)),
+                   metadata = MetaData();
```

Comment:
Unused argument. Anyway for now `MetaData` is purely internal, like `Index`, so it shouldn't appear here.

Reply:
okay this part is still a bit confusing for me. but i think this makes sense.